plugins/toolbar/plugin.js

```diff
@@ -535,6 +535,10 @@
 
 					// Ignore items that are configured to be removed.
 					if ( !removeButtons || CKEDITOR.tools.indexOf( removeButtons, name ) == -1 ) {
+						// Option to define UI element in `config.toolbar` (#2091).
+						if ( item.type && item.type in editor.ui._.handlers ) {
```

Comment:
This feature of supporting type property is nowhere to be found in the API docs. It needs to be reflected in API docs.